csharp/Adapter/Microsoft.Spark.CSharp/Network/ByteBufChunk.cs

```diff
@@ -286,6 +288,7 @@ private void Dispose(bool disposing)
                 return;
             }
 
+            logger.LogDebug("Disposing ByteBufChunk [{0}].", RuntimeHelpers.GetHashCode(this).ToString("X"));
```

Comment:
The HashCode will not change not only in all the threads of the process, but also the processes every time started.
So, if you want to add a log line and print extra info, maybe it's more useful to print the object other than the invariant HashCode.